src/store/project.ts

```diff
@@ -558,21 +651,28 @@ const audioItemSchema = z.object({
     styleId: styleIdSchema,
   }),
   query: audioQuerySchema.optional(),
-  presetKey: z.string().optional(),
+  presetKey: presetKeySchema.optional(),
   morphingInfo: morphingInfoSchema.optional(),
 });
 
 const projectSchema = z.object({
   appVersion: z.string(),
-  // description: "Attribute keys of audioItems.",
-  audioKeys: z.array(audioKeySchema),
-  // description: "VOICEVOX states per cell",
-  audioItems: z.record(audioKeySchema, audioItemSchema),
+  isOpenSongEditor: z.boolean(),
+  talk: z.object({
+    // description: "Attribute keys of audioItems.",
+    audioKeys: z.array(audioKeySchema),
+    // description: "VOICEVOX states per cell",
+    audioItems: z.record(audioKeySchema, audioItemSchema),
+  }),
+  sing: z.object({
+    tpqn: z.number(),
+    tempos: z.array(tempoSchema),
+    timeSignatures: z.array(timeSignatureSchema),
+    tracks: z.array(trackSchema),
+    // storeではmapになっているが、mapをそのまま保存すると普通のオブジェクトに
+    // なってしまい、パースに失敗するのであえてrecordにしている
```

Comment:
https://zod.dev/?id=maps
z.mapが使えそう？

Reply:
プロジェクトファイルは`JSON.stringify`で書き出されていますが、`JSON.stringify`はmapを書き出すことが出来ない([空のオブジェクトに変換される](https://solutionware.jp/2021/10/15/json-stringify-json-parse-%E3%81%A7-map-set-%E7%AD%89%E3%82%92%E6%89%B1%E3%81%86/))ため、`object`に変換する必要があります。
プロジェクトファイルをschemaにかける際、`phrases`は`map`ではなく`object`になっているので、`z.map`でバリデーションしようとすると`invalid type`になるので、`z.record`を使うのが正しいです。

コメントについては、私も今の今まで勘違いしていて、Mapが勝手にobjectに変換されて保存されていると思っていたのですが、正確にはMapが保存できなくて空のObjectに変換されてしまう、が正しいですね...
コメントは以下のように書きなおそうと思います。
```suggestion
    // storeではmapになっているが、`JSON.stringify`や`JSON.parse`はmapを保存・読み込みすることが出来ないので、
    // objectに変換する必要がある。よって、`z.map`ではなく`z.record`を使っている。
```

Reply:
あーなるほど。理解しました